subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodeTarget.java

```diff
@@ -130,4 +134,12 @@ public ConfigurableFileCollection getHeaderSearchPaths() {
     public ConfigurableFileCollection getCompileModules() {
         return compileModules;
     }
+
```

Comment:
The `headerSearchPath`,  `compileModules` and `sources` file collections should all just carry the task dependency information. These task dependencies shouldn't be tracked separately, otherwise they get out of sync, for example when someone adds something to the header file set but we don't update the task dependencies. Or when someone queries the files but forgets to query the task dependencies.

Maybe just merge what we have here and we can simplify it later.

Reply:
I agree with you on this. When I discuss this with Sterling, we figure that changing the model to correctly add `@Input` annotation or changing the task to take individual properties would have been too risky of going down a rabbit hole. I will open an issue for this.

Reply:
Issue created: https://github.com/gradle/gradle-native/issues/332.